### PR TITLE
Move writer into libs

### PR DESCRIPTION
**Pull request by JasonMarechal25**

### CODE REVIEW

**Comment by flomnes** [comment]
Please resolve conflicts

File: src/libs/antares/benchmarking/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        PRIVATE
+        ${CMAKE_SOURCE_DIR}/ext/yuni/src
+        ${CMAKE_SOURCE_DIR}/libs
+        )
+
+target_link_libraries(benchmarking
+        PRIVATE
+        yuni-static-core
+        inifile
+        result_writer
```
Comment (sylvlecl) [inline src/libs/antares/benchmarking/CMakeLists.txt]: I cannot find where the writer is used in the source files of the lib ?

Comment (flomnes) [reply src/libs/antares/benchmarking/CMakeLists.txt]: To measure the time spent on writing zip files (formerly, to measure the time waiting for the mutex to be unlocked)

Comment (sylvlecl) [reply src/libs/antares/benchmarking/CMakeLists.txt]: I mean that in the benchmarking lib source files, we don't include the writer headers --> it seems to me that it's not used inside this lib ?

File: src/libs/antares/benchmarking/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,27 @@
+set(SRC_BENCHMARKING
```
Comment (sylvlecl) [inline src/libs/antares/benchmarking/CMakeLists.txt:1]: Note: we can probably also use this PR to remove the unused source file `file_writer.cpp` in the benchmarking directory.

File: src/libs/antares/benchmarking/CMakeLists.txt (hunk 1 above)
Comment (sylvlecl) [reply src/libs/antares/benchmarking/CMakeLists.txt]: It builds without the dependency, on my side.

**Comment by flomnes** [comment]
Note : there are many conflicts, please resolve them

File: src/libs/antares/benchmarking/CMakeLists.txt (hunk 1 above)
Comment (flomnes) [reply src/libs/antares/benchmarking/CMakeLists.txt]: That's surprising !

**Comment by JasonMarechal25** [comment]
> Note : there are many conflicts, please resolve them

No more

File: src/libs/antares/benchmarking/CMakeLists.txt (hunk 2 above)
Comment (JasonMarechal25) [reply src/libs/antares/benchmarking/CMakeLists.txt:1]: Yep

File: src/libs/antares/benchmarking/CMakeLists.txt (hunk 1 above)
Comment (JasonMarechal25) [reply src/libs/antares/benchmarking/CMakeLists.txt]: Dependency is more of a "header" deps through study.h

**Comment by sylvlecl** [comment]
Closing this one in favor of https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/1548, which is an improved version of this.

Also, this was not just anymore about moving the writer, it makes more sense to have a more global PR about build improvements.

**Pull request closed without merging**